File: 2022/kotlin/day6.kt
import java.io.File

fun main() {
    val sample = "mjqjpqmgbljsphdztnvjfqwrcgsmlb"
    val input = File("input/day6.txt").bufferedReader().readLine()
    val messageSize = 14

    val answer = messageSize.until(input.length).first { i ->
        input.subSequence(i - messageSize, i).toSet().size == messageSize
    }

    println(answer)
}

File: 2022/swift/day6.swift
func day6() {
    let sample = Array("mjqjpqmgbljsphdztnvjfqwrcgsmlb")
    let n = 4
    let answer = (n..<sample.count).first(where: { Set(sample[$0-n..<$0]).count == n })
    print(answer as Any)
}

day6()